modules/utxo-lib/src/testutil/psbt.ts

```diff
@@ -0,0 +1,188 @@
+import * as assert from 'assert';
+import {
+  createOutputScriptP2shP2pk,
+  ScriptType,
+  ScriptType2Of3,
+  scriptTypeP2shP2pk,
+  scriptTypes2Of3,
+} from '../bitgo/outputScripts';
+import {
+  addReplayProtectionUnspentToPsbt,
+  addWalletOutputToPsbt,
+  addWalletUnspentToPsbt,
+  createPsbtForNetwork,
+  getExternalChainCode,
+  isWalletUnspent,
+  KeyName,
+  RootWalletKeys,
+  Unspent,
+  UtxoPsbt,
+} from '../bitgo';
+import { Network } from '../networks';
+import { mockReplayProtectionUnspent, mockWalletUnspent } from './mock';
+import { getInternalChainCode } from '../bitgo';
+
+/**
+ * input script type and value.
+ * use p2trMusig2 for p2trMusig2 script path.
+ * use taprootKeyPathSpend for p2trMusig2 key path.
+ */
+export type InputScriptType = ScriptType | 'taprootKeyPathSpend';
+export type OutputScriptType = ScriptType2Of3;
+
+/**
+ * output script type and value
+ */
+export interface Input {
+  scriptType: InputScriptType;
+  value: bigint;
+}
+
+/**
+ * set isInternalAddress=true for internal output address
+ */
+export interface Output {
+  scriptType: OutputScriptType;
+  value: bigint;
+  isInternalAddress?: boolean;
+}
+
+/**
+ * array of supported input script types.
+ * use p2trMusig2 for p2trMusig2 script path.
+ * use taprootKeyPathSpend for p2trMusig2 key path.
+ */
+export const inputTypes = [...scriptTypes2Of3, 'taprootKeyPathSpend', scriptTypeP2shP2pk] as const;
+
+/**
+ * array of supported output script types.
+ */
+export const outputTypes = scriptTypes2Of3;
+
+/**
+ * create unspent object from input script type, index, network and root wallet key.
+ */
+export function toUnspent(
+  input: Input,
+  index: number,
+  network: Network,
+  rootWalletKeys: RootWalletKeys
+): Unspent<bigint> {
+  if (input.scriptType === 'p2shP2pk') {
+    return mockReplayProtectionUnspent(network, input.value, { key: rootWalletKeys['user'], vout: index });
+  } else {
+    const chain = getInternalChainCode(input.scriptType === 'taprootKeyPathSpend' ? 'p2trMusig2' : input.scriptType);
+    return mockWalletUnspent(network, input.value, {
+      chain,
+      vout: index,
+      keys: rootWalletKeys,
+      index,
+    });
+  }
+}
+
+/**
+ * returns signer and cosigner names for InputScriptType.
+ * user and undefined as signer and cosigner respectively for p2shP2pk.
+ * user and backup as signer and cosigner respectively for p2trMusig2.
+ * user and bitgo as signer and cosigner respectively for other input script types.
+ */
+export function getSigners(inputType: InputScriptType): { signerName: KeyName; cosignerName?: KeyName } {
```

Comment:
To be able to test MuSig2 signing, shouldn't this return `bitgo` for all types other than `p2shP2pk`?

Reply:
Yes. Thats how it works. Only `p2trMusig2` will return `backup`for cosigner since we assume it as `script path` (for simplicity). In hsm-api test suite, we will skip `p2trMusig2`.  Script type `taprootKepPathSpend` is `p2trMusig2 key path`.